# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ThirdPartyUtil.h
# Commit: bfe21f274ed5
# Full Hash: bfe21f274ed5a4630a3503bcb875fed04f90533c
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-04-16 03:00:04
# Regressor Bug: 1594529
# File Overlap Count: 1
# Description:
#   Bug 1594529 - Create LoadInfo for subdocuments directly in parent process with DocumentChannel. r=mattwoodrow,nika
#   
#   Currently, with Fission enabled we are not able to create a proper LoadInfo
#   object when doing a subdocument load because we do not have access to a loading
#   context if the load is happening inside of an OOP frame. To solve this problem,
# ==============================================================================

diff -r 29ff609063c2 -r bfe21f274ed5 dom/base/ThirdPartyUtil.h
--- a/dom/base/ThirdPartyUtil.h	Wed Apr 15 18:53:22 2020 +0000
+++ b/dom/base/ThirdPartyUtil.h	Wed Apr 15 18:53:06 2020 +0000
@@ -14,6 +14,7 @@
 #include "nsEffectiveTLDService.h"
 #include "nsString.h"
 #include "nsPIDOMWindow.h"
+#include "mozilla/dom/WindowGlobalParent.h"
 
 class nsIURI;
 class nsPIDOMWindowOuter;
@@ -28,6 +29,9 @@
   static void Startup();
   static ThirdPartyUtil* GetInstance();
 
+  nsresult IsThirdPartyGlobal(mozilla::dom::WindowGlobalParent* aWindowGlobal,
+                              bool* aResult);
+
  private:
   ~ThirdPartyUtil();
 